src/network/wait-online/manager.c

```diff
@@ -381,13 +407,102 @@ static int manager_network_monitor_listen(Manager *m) {
         return 0;
 }
 
+static int on_dns_configuration_event(
+                sd_varlink *link,
+                sd_json_variant *parameters,
+                const char *error_id,
+                sd_varlink_reply_flags_t flags,
+                void *userdata) {
+
+        Manager *m = ASSERT_PTR(userdata);
+        sd_json_variant *configurations = NULL, *v = NULL;
+        int r;
+
+        assert(link);
+
+        if (error_id) {
```

Comment:
I think we should reconnect the server when disconnected, for the case that resolved is restarted.

Or, maybe better that resolved serializes connections on restart.

Reply:
Serializing connections on restart seems like the better solution to me. Could I handle that in a separate follow-up PR?

Reply:
Yeah. I think it's ok to implement that later.

Reply:
FWIW it also gives up immediately if resolved is down for whatever reason so it doesn't actually take the timeout into account in all cases.